mm2src/coins/utxo/utxo_builder/utxo_coin_builder.rs

```diff
@@ -230,15 +230,15 @@ where
     let key_pair = priv_key_policy.activated_key_or_err()?;
     let addr_format = builder.address_format()?;
     let my_address = Address {
-        prefix: conf.pub_addr_prefix,
-        t_addr_prefix: conf.pub_t_addr_prefix,
+        prefixes: conf.address_prefixes.p2pkh.clone(),
         hash: AddressHashEnum::AddressHash(key_pair.public().address_hash()),
         checksum_type: conf.checksum_type,
         hrp: conf.bech32_hrp.clone(),
         addr_format,
+        script_type: AddressScriptType::P2PKH,
```

Comment:
This can be `AddressScriptType::P2WPKH` depending on the address format. Without fixing this, the change I suggested in the comment above will not work.

Reply:
I think even better would be to use builder:
```
    let my_address = AddressBuilder {
        prefixes: conf.address_prefixes.p2pkh.clone(),
        hash: AddressHashEnum::AddressHash(key_pair.public().address_hash()),
        checksum_type: conf.checksum_type,
        hrp: conf.bech32_hrp.clone(),
        addr_format,
        is_pay2pk: AddressScriptType::P2PKH,
    }.build();```

Reply:
I agree, Builder pattern is a good fit for this case, we should have `AddressBuilder::new()` function to initiate the building process, with setter functions for the address parameters.

Reply:
added AddressBuilder in [87c438d](https://github.com/KomodoPlatform/komodo-defi-framework/pull/1960/commits/87c438d11d2f42244943b846d6b1fa29b358d9a4) with extra creation data checks (which allowed to find an issue in swaps with segwit [a38ed54](https://github.com/KomodoPlatform/komodo-defi-framework/pull/1960/commits/a38ed54a911a2f9c2afe8ee6d1078c6a71126bb8): Segwit is not supported in swaps so addr_format must be converted from segwit to legacy here)